app/src/main/java/de/kabiskac/weekdaybug/ui/main/Fragment1.kt

```diff
@@ -24,9 +25,11 @@ class Fragment1 : Fragment() {
             inflater,
             R.layout.fragment1, container, false
         )
-        viewModel = ViewModelProvider(this).get(MainViewModel::class.java)
+        viewModel = ViewModelProvider(requireActivity()).get(MainViewModel::class.java)
 
-        binding.viewmodel = viewModel
+        handler.post {
+            binding.viewmodel = viewModel
+        }
```

Comment:
This was the root of the the issue. `MaterialDayPicker` restores it's own instance state which happens after it's initialized. By delaying attaching your ViewModel to the binding it gives `MaterialDayPicker` a chance to restore its state before the latest state from the ViewModel is bound.